src/main/java/de/tum/in/www1/artemis/repository/SubmissionRepository.java

```diff
@@ -442,6 +444,23 @@ default Submission findByIdWithResultsElseThrow(long submissionId) {
         return findWithEagerResultsAndAssessorById(submissionId).orElseThrow(() -> new EntityNotFoundException("Submission", +submissionId));
     }
 
+    /**
+     * Gets all Submissions which are submitted
+     *
+     * @param exerciseId the ID of the exercise
+     * @param pageable   the pagination information for the query
+     * @return Set of Submissions
+     */
+    Page<Submission> findByParticipation_ExerciseIdAndSubmittedIsTrue(long exerciseId, Pageable pageable);
```

Comment:
You are using it in the `AthenaSubmissionSendingService` for sending the submitted submission, but can submitted be true for multiple submissions? I think so, don't we need to send only the latest that is getting graded? Same with the code below

Reply:
Oh, that's a great catch! I corrected this and I will explicitly test that only the newest submissions are sent on the test server.